### PR TITLE
Manifest remove, push

### PR DESCRIPTION
**Pull request by QiWang19**
Implements podman manifest remove and podman manifest push.

Signed-off-by: Qi Wang <qiwan@redhat.com>

### CODE REVIEW

File: cmd/podman/manifest/push.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+func push(cmd *cobra.Command, args []string) error {
+	if err := buildahcli.CheckAuthFile(manifestPushOpts.Authfile); err != nil {
+		return err
+	}
+	listImageSpec := args[0]
+	destSpec := args[1]
+	if listImageSpec == "" {
+		return errors.Errorf(`Invalid image name "%s"`, listImageSpec)
```
Comment (rhatdan) [inline cmd/podman/manifest/push.go:57]: errors begin lower case

File: cmd/podman/manifest/push.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	}
+	if destSpec == "" {
+		return errors.Errorf(`Invalid image name "%s"`, destSpec)
```
Comment (rhatdan) [inline cmd/podman/manifest/push.go:60]: return errors.Errorf(`Invalid destination "%s"`, destSpec)

File: cmd/podman/manifest/remove.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+func remove(cmd *cobra.Command, args []string) error {
+	listImageSpec := args[0]
+	instanceSpec := args[1]
+	if listImageSpec == "" {
+		return errors.Errorf(`Invalid image name "%s"`, listImageSpec)
+	}
+	if instanceSpec == "" {
+		return errors.Errorf(`Invalid instance "%s"`, instanceSpec)
```
Comment (rhatdan) [inline cmd/podman/manifest/remove.go:39]: Do users know what an instance is?

Comment (rhatdan) [reply cmd/podman/manifest/remove.go:39]: lower case

File: docs/source/markdown/podman-manifest-push.1.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
+% podman-manifest-push(1)
+
+## NAME
+podman\-manifest\-push - Push a manifest list or image index to a registry
+
+## SYNOPSIS
+**podman manifest remove** *listnameorindexname* *imagemanifestdigest*
```
Comment (rhatdan) [inline docs/source/markdown/podman-manifest-push.1.md:7]: Remove?

File: pkg/domain/infra/abi/manifest.go [hunk 5]
```diff
@@ -99,3 +106,68 @@ func (ir *ImageEngine) ManifestAdd(ctx context.Context, opts entities.ManifestAd
 	}
 	return listID, nil
 }
+
+// ManifestRemove removes specified digest from the specified manifest list
+func (ir *ImageEngine) ManifestRemove(ctx context.Context, names []string) (string, error) {
+	instanceDigest, err := digest.Parse(names[1])
+	if err != nil {
+		return "", errors.Errorf(`Invalid instance "%s": %v`, names[1], err)
```
Comment (rhatdan) [inline pkg/domain/infra/abi/manifest.go:114]: lower case

Comment (rhatdan) [reply pkg/domain/infra/abi/manifest.go:114]: Do you users know what an instance is?

**Comment by QiWang19** [comment]
@jwhonce @mheon @rhatdan PTAL

File: cmd/podman/manifest/push.go [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+	flags.StringVar(&manifestPushOpts.CertDir, "cert-dir", "", "use certificates at the specified path to access the registry")
+	flags.StringVar(&manifestPushOpts.Creds, "creds", "", "use `[username[:password]]` for accessing the registry")
+	flags.StringVar(&manifestPushOpts.DigestFile, "digestfile", "", "after copying the image, write the digest of the resulting digest to the file")
+	flags.StringVarP(&manifestPushOpts.Format, "format", "f", "", "manifest type (oci or v2s2) to attempt to use when pushing the manifest list (default is manifest type of source)")
+	flags.BoolVarP(&manifestPushOpts.RemoveSignatures, "remove-signatures", "", false, "don't copy signatures when pushing images")
+	flags.StringVar(&manifestPushOpts.SignBy, "sign-by", "", "sign the image using a GPG key with the specified `FINGERPRINT`")
+	flags.BoolVar(&manifestPushOpts.TlsVerify, "tls-verify", true, "require HTTPS and verify certificates when accessing the registry")
+	flags.BoolVarP(&manifestPushOpts.Quiet, "quiet", "q", false, "don't output progress information when pushing lists")
+	if registry.IsRemote() {
+		_ = flags.MarkHidden("authfile")
```
Comment (rhatdan) [inline cmd/podman/manifest/push.go:44]: authfile needs to be eventually supported in remote.  But this is fine for now.

**Comment by rhatdan** [comment]
LGTM

**Comment by QiWang19** [comment]
@TomSweeneyRedHat @jwhonce @mheon @rhatdan @TomSweeneyRedHat  PTAL

**Comment by rhatdan** [comment]
@baude @mheon @vrothberg @giuseppe @jwhonce @TomSweeneyRedHat PTAL

**Comment by QiWang19** [comment]
can this get an approval label?

**Pull request merged**